test/request-timeout.js

```diff
@@ -342,3 +342,41 @@ test('Request options overrides global option', { skip: 'buggy' }, (t) => {
     clock.tick(50)
   })
 })
+
+test('client.destroy should cancel the timeout', (t) => {
+  t.plan(2)
+
+  const server = createServer((req, res) => {
+    res.end('hello')
+  })
+  t.teardown(server.close.bind(server))
+
+  server.listen(0, () => {
+    const client = new Client(`http://localhost:${server.address().port}`)
+
+    client.request({ path: '/', method: 'GET', requestTimeout: 100 }, (err, response) => {
+      t.ok(err instanceof errors.ClientDestroyedError)
+    })
+
+    client.destroy(err => t.error(err))
+  })
+})
+
+test('client.close should cancel the timeout', (t) => {
```

Comment:
close should not cancel timeout. It should gracefully wait for all enqueued requests to finish and not allow any new ones.

Reply:
I think it should (if the request has not been sent yet, like in this test), otherwise the callback will be called again with `RequestTimeoutError`.

Reply:
>  otherwise the callback will be called again with RequestTimeoutError,

There is some other bug here then. `close` should wait for everything finish normally, i.e. no errors unless something unexpected happened.

Reply:
Do you have time to track it down?

Reply:
Comment out this test and finish up the rest and I will sort it out later.

Reply:
`close()` should finish things normally (or have them timeout).  `destroy()` should cancel it.

Reply:
I have fixed this. However, please see the commit below in regard to error propagation to close.